### ISSUE TITLE
Not parsing #if correctly in C++

### ISSUE THREAD

**Issue by stickM4N** [doxygen/doxygen#8940]
Doxygen is not recognizing correclty C++ #if preprocessor directive.

When using 
```c++
#if not defined(SOMETHING)
    ...
#endif
```
Or something like
```c++
#if true
    ...
#endif
```
results in a false condition and the docs of that section are not documented.

It should work as #ifndef or #if 1, and they are not...

So in my humble opinion &&, || and ! are not working propertly with their equivalents nor true and false substitutions.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8941

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by stickM4N** [comment]
Wow. Great response time 😅.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
